### PR TITLE
Read max file size from capabilities

### PR DESCRIPTION
**Pull request by dbrgn**
Once we've integrated flow controlled data channels (#769), the data channels don't die anymore when sending large files. Thus, we could remove the artificial 15 MiB size limit when sending files through Threema Web.

However, due to the encrypt-then-chunk scheme, older phones with little memory might crash due to an OOM situation when receiving large files from Threema Web. Therefore we might still want to be able to control the max file size that can be sent until we've switched to the chunk-then-encrypt scheme.

We have two possibilities:

**a) Just keep the limitation**

- Upsides: Simple
- Downsides: Even though both Threema Web and newer devices might be fully capable of sending 50 MiB or 100 MiB files, we deny that possibility because some older devices might crash.

**b) Determine the limit through the app capabilities message**

-  Upsides: People with newer phones can already enjoy the benefits of sending large files.
- Downsides: We have another parameter in the capabilities message. However, the field is optional (falling back to 15 MiB when not set) and can be removed again once we have implemented the ack protocol.
  - Current app: Limit stays at 15 MiB
  - App which sends the maxWebrtcFileSize field: Indicates to Threema Web that it can handle larger files depending on available RAM
- App with an ack protocol implementation: Does not send the maxWebrtcFileSize field, Threema Web drops limits when the connection was established using the ack protocol.

Opinions, @lgrahl?

Blocked by #769.

### CODE REVIEW

**Comment by lgrahl** [comment]
First of all, we should mention that a) would not be permanent. It would be resolved eventually with #498 for Android.

b) will be just as brittle as the 15 MiB heuristic applied today, so I honestly don't see much value in it. A person having a phone with 4 GiB RAM will probably also use more RAM, so there's no guarantee a crash can be avoided. 50 MiB will still be at least 150 MiB in RAM due to excessive amounts of copying and the lack of a streaming implementation.

I don't mind going with either... however, I'm against adding a new *capabilities* field as explained in-line.

File: src/services/webclient.ts [hunk 1]
```diff
@@ -3084,6 +3087,7 @@ export class WebClientService {
             capabilities: {
                 maxGroupSize: getOrDefault<number>(data.capabilities.maxGroupSize, 50),
                 maxFileSize: getOrDefault<number>(data.capabilities.maxFileSize, 50 * 1024 * 1024),
+                maxWebrtcFileSize: getOrDefault<number>(data.capabilities.maxWebrtcFileSize, 15 * 1024 * 1024),
```
[the remark this answers is not in the document]
Comment (dbrgn) [reply src/services/webclient.ts]: > I don't see a point in adding a new field. The app can just set maxFileSize accordingly.

Hmm, `maxFileSize` has different semantics (it's the threema protocol size limit), but if you have an idea how to combine the two error messages we could do that. Right now there is a distinction between "Files larger than X MiB cannot be sent" vs "Files larger than X MiB cannot currently be sent through Threema Web" which I think is important.

However, I don't know how we could introduce such a change in a backwards compatible way, that it still works properly both with old app + new webclient and new app + old webclient.

> maxWebrtcFileSize would also be the wrong terminology here as OOM is completely unrelated to WebRTC

Yup, but it only happens with the current WebRTC task, the Relayed Data task is unaffected. Of course the WebRTC task v2 would be a different story. If you have a better naming idea, let me know.

**Comment by dbrgn** [comment]
> A person having a phone with 4 GiB RAM will probably also use more RAM, so there's no guarantee a crash can be avoided.

Yes, but a phone with 4 GiB RAM can suspend a few background tasks and then it should have enough RAM for 150 MiB of file data. If you have a phone with 1 GiB, then it will be hard to free 150 MiB of RAM.

File: src/services/webclient.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lgrahl) [reply src/services/webclient.ts]: How about `maxFileSizeUntilAckProtocol`?

**Comment by dbrgn** [comment]
I'm unsure if we should continue with this. @lgrahl do you have a rough estimate how long it might take until we can implement streaming? Probably at least half a year, right?

**Comment by lgrahl** [comment]
It will be hard to explain why person A can send 50 MiB messages via the web client while B cannot. I think we should go with a) for the time being and fix this properly asap.

**Pull request closed without merging**